src/differentials/thunks.jl

```diff
@@ -13,6 +13,10 @@ end
     return element, (val, new_state)
 end
 
+Base.:(==)(a::AbstractThunk, b::AbstractThunk) = unthunk(a) == unthunk(b)
```

Comment:
There are some special case we might be able to optimize the positive cases like `a===b` to avoid unthunking.
idk if worth it though